apps/cowswap-frontend/src/modules/orders/containers/ExpiredOrderNotification/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback } from 'react'
+
+import { OnExpiredOrderPayload } from '@cowprotocol/events'
+import { TokenInfo } from '@cowprotocol/types'
+
+import { EnhancedTransactionLink } from 'legacy/components/EnhancedTransactionLink'
+import { HashType } from 'legacy/state/enhancedTransactions/reducer'
+import { useOrder } from 'legacy/state/orders/hooks'
+
+import { OrderLinkWrapper } from '../../pure/commonStyled'
+import { OrderSummary } from '../../pure/OrderSummary'
+
+export interface ExpiredOrderNotificationProps {
+  payload: OnExpiredOrderPayload
+  onToastMessage(message: string): void
+}
+
+export function ExpiredOrderNotification(props: ExpiredOrderNotificationProps) {
+  const {
+    payload: {
+      chainId,
+      order: { uid: orderUid },
+    },
+    onToastMessage,
+  } = props
+
+  const ref = useCallback(
+    (node: HTMLDivElement) => {
+      if (node) onToastMessage(node.innerText)
+    },
+    [onToastMessage]
+  )
```

Comment:
this pattern seems a bit obscure to me. If we are using it, at least we could make a reusable hook

Reply:
Agree with you, refactored in https://github.com/cowprotocol/cowswap/pull/3934